### DISCUSSION TITLE
Support for private container registries (Bicep modules)?

### DISCUSSION THREAD

**Discussion by SvenAelterman** [Azure/template-analyzer#354] [Q&A]
Hi,

It appears that Template Analyzer fails on analyzing Bicep files that reference modules in private container registry. This is probably not surprising, as there appears to be no way to provide Template Analyzer with a credential to authenticate against the private registry.

Is this a known issue? Is there a workaround?

PS: this is coming from a use case that involves Defender for Cloud DevOps Security.

PPS: scanning of the Bicep modules that are pushed to the registry is already in place.

**Comment by SvenAelterman** [comment]
Anyone?

**Comment by craigofnz** [comment]
If it cannot access a private registry directly, it would be nice if it could support using the local bicep modules cache in $Env:BICEP_CACHE_DIRECTORY, such that it could evaluate the current main.bicep, parameters and any upstream modules included main.bicep and other referenced bicep files or modules.
